### PR TITLE
fix: remove short flag in install.sh

### PR DESCRIPTION
**Pull request by Qaplagzy**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
> Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.

To maintain compatibility and support for the current version of kubectl.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/4164
**Does this PR introduce a user-facing change?**:

```release-note
NONE

```

### CODE REVIEW

**Review by wbc6080 — COMMENTED** [review]
A small question, in which version was the kubectl deprecation --short feature proposed? Since the kubeedge master version supports three versions of kubernetes: 1.26, 1.27 and 1.28, if this feature was introduced in 1.28, but the user's kubernetes is version 1.26, will deleting --short cause problems?

**Review by Shelley-BaoYue — COMMENTED** [review]
There are something wrong with ci and I have solved it and merge it into master.  Please rebase the new master branch and push again. Thx : ）

File: hack/lib/install.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ function check_kubectl {
     echo "kubectl not installed, exiting."
     exit 1
   else
-    echo -n "found kubectl, " && kubectl version --short --client
+    echo -n "found kubectl, " && kubectl version --client
```
[the remark this answers is not in the document]
Comment (Qaplagzy) [reply hack/lib/install.sh:25]: it throws the error log
`checking kubectl
found kubectl, error: unknown flag: --short
See 'kubectl version --help' for usage.`

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm

**Pull request merged**